plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java

```diff
@@ -626,19 +624,56 @@ public void testDate(ZoneId sessionZone)
                 .execute(getQueryRunner(), session, clickhouseCreateAndInsert("tpch.test_date"));
     }
 
-    @Test
-    public void testUnsupportedDate()
+    @Test(dataProvider = "clickHouseDateMinMaxValuesDataProvider")
+    public void testClickHouseDateMinMaxValues(String date)
+    {
+        SqlDataTypeTest dateTests = SqlDataTypeTest.create()
+                .addRoundTrip("date", format("DATE '%s'", date), DATE, format("DATE '%s'", date));
+
+        for (Object[] timeZoneIds : sessionZonesDataProvider()) {
+            Session session = Session.builder(getSession())
+                    .setTimeZoneKey(TimeZoneKey.getTimeZoneKey(((ZoneId) timeZoneIds[0]).getId()))
+                    .build();
+            dateTests
+                    .execute(getQueryRunner(), session, clickhouseCreateAndInsert("tpch.test_date"))
+                    .execute(getQueryRunner(), session, trinoCreateAsSelect(session, "test_date"))
+                    .execute(getQueryRunner(), session, trinoCreateAsSelect("test_date"))
+                    .execute(getQueryRunner(), session, trinoCreateAndInsert(session, "test_date"))
+                    .execute(getQueryRunner(), session, trinoCreateAndInsert("test_date"));
+        }
+    }
+
+    @DataProvider
+    public Object[][] clickHouseDateMinMaxValuesDataProvider()
+    {
+        return new Object[][] {
+                {"1970-01-01"}, // min value in ClickHouse
+                {"2106-02-07"}, // max value in ClickHouse
+        };
+    }
+
+    @Test(dataProvider = "unsupportedClickHouseDateValuesDataProvider")
+    public void testUnsupportedDate(String unsupportedDate)
     {
+        String minSupportedDate = (String) clickHouseDateMinMaxValuesDataProvider()[0][0];
+        String maxSupportedDate = (String) clickHouseDateMinMaxValuesDataProvider()[1][0];
+
         try (TestTable table = new TestTable(getQueryRunner()::execute, "test_unsupported_date", "(dt date)")) {
             assertQueryFails(
-                    format("INSERT INTO %s VALUES (DATE '1969-12-31')", table.getName()),
-                    "Date must be between 1970-01-01 and 2106-02-07 in ClickHouse: 1969-12-31");
-            assertQueryFails(
-                    format("INSERT INTO %s VALUES (DATE '2106-02-08')", table.getName()),
-                    "Date must be between 1970-01-01 and 2106-02-07 in ClickHouse: 2106-02-08");
+                    format("INSERT INTO %s VALUES (DATE '%s')", table.getName(), unsupportedDate),
+                    format("Date must be between %s and %s in ClickHouse: %s", minSupportedDate, maxSupportedDate, unsupportedDate));
```

Comment:
In addition to this we should probably INSERT using ClickHouse and then read back the value and assert that the values don't match - this will ensure that the min/max ranges we are using are up to date and match the implementation on ClickHouse side.